ctsit/nacculator
# Update in-line comments and docstrings

This change will implement updates to the in-line code and docstrings attached to functions within the code. The updates on this branch are geared toward the benefit of NACCulator developers rather than users. It does not change the way NACCulator operates in any way, other than updating the title of a filter that was misleading and confusing ("fix_visitdate" to "fix_visitnum", since the filter did not actually affect the visit date at all). 

This change was made because NACCulator is a large and complicated machine, and good documentation will allow people to more quickly see how it works when they develop new functionalities. 


## Verification

After activating the python 3 virtual environment, and making sure you have the config file set up with a REDCap API token...

1. From the command line, run `python3 -m unittest` to verify all unit tests work.
2. Next, on the command line, run `nacculator_filters nacculator_cfg.ini` to automatically download the data from our REDCap project and run all filters.
3. Run `redcap2nacc -ivp <run_01-23-2023/final_Update.csv >run_01-23-2023/ivp_nacc_complete.txt 2>run_01-23-2023/ivp_errors.txt` (replacing 01-23-2023 with the current date) (or use whichever flags you like: -fvp, -tfp, -ivp -lbdsv, and so on)
4. The output should look like this: Normal NACCulator output, with an ivp_nacc_complete.txt file and ivp_errors.txt file.
5. If you pass data with errors, the output should look like this: NACCulator will catch and log the error as expected.
6. Check within the code to see that each .py file has functions with docstrings and comments that make sense and illustrate what the program is doing.
7. If you have any questions about any of the docstrings or operations that weren't explicitly described, please tell me here and I will add explanations to those parts of the code.

- Verify the thing does this: An accurate description of NACCulator's inner workings.
- Verify the thing does not do that: Change the way NACCulator works in any way.


## Affirmations

All of these should have a check by them. Any exception requires an explanation.
